src/compiler/compile/render_ssr/index.ts

```diff
@@ -52,6 +54,58 @@ export default function ssr(
 		})
 		.filter(Boolean);
 
+	// instrument get/set store value
+	if (component.ast.instance) {
+		let scope = component.instance_scope;
+		const map = component.instance_scope_map;
+
+		walk(component.ast.instance.content, {
+			enter(node: Node) {
+				if (map.has(node)) {
+					scope = map.get(node);
+				}
+			},
+			leave(node: Node) {
+				if (map.has(node)) {
+					scope = scope.parent;
+				}
+
+				if (node.type === 'AssignmentExpression' || node.type === 'UpdateExpression') {
+					const assignee = node.type === 'AssignmentExpression' ? node.left : node.argument;
+					const names = new Set(extract_names(assignee));
+
+					const vars = Array.from(names)
+						.filter(name => {
+							const owner = scope.find_owner(name);
+							return !owner || owner === component.instance_scope;
+						})
+						.map(name => component.var_lookup.get(name))
+						.filter(variable =>	{
+							return variable && 
+								!variable.hoistable &&
+								!variable.global &&
+								!variable.module &&
+								(
+									variable.subscribable || variable.name[0] === '$'
+								);
+						})
+						.map(variable => {
+							if (variable.subscribable) {
+								return x`${'$' + variable.name} = @get_store_value(${variable.name})`;
+							}
+							if (variable.name[0] === '$') {
+								return x`${variable.name.slice(1)}.set(${variable.name})`;
+							}
+						});
+
+					if (vars.length) {
+						this.replace(x`@identity(${node}, ${vars})`);
```

Comment:
Is there a reasonable way to optimize this and not call `identity()` if the value is discarded? Actually, I'm not entirely sure why `identity` is needed at all. Could you instead use a sequence expression where `${node}` occurs last?

Reply:
Nice, this looks better I think.